src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1250,13 +1249,17 @@ QVector<PeerInfo> TorrentImpl::peers() const
 
 QBitArray TorrentImpl::pieces() const
 {
-    QBitArray result(m_nativeStatus.pieces.size());
-    for (int i = 0; i < result.size(); ++i)
+    if (m_pieces.isEmpty())
```

Comment:
Previously it always get the latest bitfield from libtorrent and now it returns cached data.
This changes the api usage, is it correct?

Reply:
> Previously it always get the latest bitfield from libtorrent and now it returns cached data.

No. Previously it returns cached (in m_nativeStatus) data in libtorrent format.